src/pageindicator/index.ts

```diff
@@ -0,0 +1,45 @@
+import pageIndicatorStyle from 'inline:./style.css';
+import { html, LitElement, TemplateResult, unsafeCSS } from 'lit';
+import { property } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { map } from 'lit/directives/map.js';
+import { range } from 'lit/directives/range.js';
+
+export class WPageIndicator extends LitElement {
+  static styles = [unsafeCSS(pageIndicatorStyle)];
+
+  /** Currently selected page (1-based index) */
+  @property({ type: Number, attribute: 'selected-page' })
+  selectedPage = 1;
+
+  /** Total number of pages */
+  @property({ type: Number, attribute: 'page-count' })
+  pageCount = 1;
+
+  render(): TemplateResult {
+    return html`
+      <div class="w-pageindicator">
+        <div class="w-pageindicator--container">
+          ${map(range(this.pageCount), (i) => {
+            const isSelected = i === this.selectedPage - 1;
+            const classes = {
+              'w-pageindicator--dot': true,
+              'w-pageindicator--selecteddot': isSelected,
+            };
+            return html`<div class="${classMap(classes)}"></div>`;
+          })}
+        </div>
+      </div>
+    `;
+  }
+}
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'w-pageindicator': WPageIndicator;
+  }
+}
+
+if (!customElements.get('w-pageindicator')) {
```

Comment:
there's a `@customElement` decorator that should replace this block

Reply:
<img width="782" height="107" alt="Screenshot 2025-08-13 at 13 28 54" src="https://github.com/user-attachments/assets/51ff6efc-965e-42e8-bb7a-0af318f95504" />

Based on the docs, the decorator doesn't do the `if (!customElements.get('w-pageindicator'))` clause which we kinda need